tests/integration/test_run_app_from_bundle.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import subprocess
+import tarfile
+
+import utils
+
+
+def test_run_app_from_bundle(test_env, default_request, tmpdir):
+    """
+    Check that downloaded bundle could be used to run the application.
+
+    Process:
+    * Send new request to Cachito API
+    * Download a bundle from the request
+    * Run go build
+    * Run the application
+
+    Checks:
+    * Check that the state of request is complete
+    * Check that the bundle is properly downloaded
+    * Check that the application runs successfully
+    """
+    response = default_request.complete_response
+    assert response.status == 200
+    assert response.data["state"] == "complete"
+
+    file_name_tar = tmpdir.join(f"download_{str(response.id)}.tar.gz")
+    file_name = tmpdir.join(f"download_{str(response.id)}")
```

Comment:
It seems like it doesn't matter, but it might be clearer if you used mkdir:
```suggestion
    bundle_dir = tmpdir.mkdir(f"download_{str(response.id)}")
```